### PR TITLE
Upgrade AWS VPC CNI plugin to 1.5

### LINKED ISSUE #927: Update Amazon VPC CNI Plugin to v1.5.0
**What happened?**
New clusters and `eksctl utils update-aws-node --name=<cluster-name> --approve` use v1.4.1 of Amazon VPC CNI Plugin. To be fair, AWS docs appear to be a bit outdated here as well from what I can tell.

**What you expected to happen?**
New cluster and `eksctl utils update-aws-node --name=<cluster-name> --approve` use v1.5.0 of Amazon VPC CNI Plugin.

**How to reproduce it?**
Create a new cluster or run the `eksctl utils update-aws-node --name=<cluster-name> --approve` command

**Anything else we need to know?**
- What OS are you using: 5.1.12-arch1-1-ARCH
- Are you using a downloaded binary or did you compile eksctl: Downloaded binary
- What type of AWS credentials are you using (i.e. default/named profile, MFA): default

**Versions**
```
$ eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.1.38"}
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"archive", BuildDate:"2019-06-20T22:55:20Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.7-eks-c57ff8", GitCommit:"c57ff8e35590932c652433fab07988da79265d5b", GitTreeState:"clean", BuildDate:"2019-06-07T20:43:03Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Logs**
Not relevant for this issue

### PR DESCRIPTION
**Pull request by martina-if**
closes #927 

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Manually tested updating aws-node addon
- [x] Manually tested creating cluster
- [x]  Wait for green light from AWS

Questions:
- When I create a cluster, the cni version is still 1.4.1. I am assuming that we don't control the version of the addon when we create a cluster; only when we update it. Is this correct?

### CODE REVIEW

**Comment by errordeveloper** [comment]
> When I create a cluster, the cni version is still 1.4.1. I am assuming that we don't control the version of the addon when we create a cluster; only when we update it. Is this correct?

Yes, and actually we will need hold this until new clusters come with 1.5.x.

File: pkg/addons/default/generate.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package defaultaddons
 
-//go:generate curl --silent --location https://github.com/aws/amazon-vpc-cni-k8s/blob/957f835f437fd76953f37e6b409ba6426bb4ce37/config/v1.4/aws-k8s-cni.yaml?raw=1 --output assets/aws-node.yaml
+//go:generate curl --silent --location https://github.com/aws/amazon-vpc-cni-k8s/blob/723dd5a2d0387477f6950b23eba6e8b586a59a70/config/v1.5/aws-k8s-cni.yaml?raw=1 --output assets/aws-node.yaml
 //go:generate curl --silent --location https://github.com/aws/amazon-vpc-cni-k8s/blob/957f835f437fd76953f37e6b409ba6426bb4ce37/config/v1.4/aws-k8s-cni-1.10.yaml?raw=1 --output assets/aws-node-1.10.yaml
```
Comment (errordeveloper) [inline pkg/addons/default/generate.go]: Is there a new one for 1.10 also?

File: pkg/addons/default/testdata/sample-1.11.json [hunk 2]
```diff
@@ -534,7 +534,7 @@
                     }
                   }
                 ],
-                "image": "602401143452.dkr.ecr.eu-west-1.amazonaws.com/amazon-k8s-cni:v1.4.1",
```
Comment (errordeveloper) [inline pkg/addons/default/testdata/sample-1.11.json]: Don't need to update these manifests, they should come from live clusters, and don't really have to be up-to-date, these are just samples.

File: pkg/addons/default/generate.go (hunk 1 above)
Comment (martina-if) [reply pkg/addons/default/generate.go]: No, I didn't see it in the repo. I wasn't sure if I should use the same file or keep 1.4 :/ What do you think?

Comment (errordeveloper) [reply pkg/addons/default/generate.go]: Ah, so perhaps they just decided to stop supporting 1.10, which is something that's meant to happen sooner or later in general.

**Comment by patstrom** [comment]
https://aws.amazon.com/about-aws/whats-new/2019/07/aws-vpc-cni-version-150-now-default-for-amazon-eks-clusters/

v1.5.0 is now the default for new clusters!

**Comment by martina-if** [comment]
Thanks @patstrom !

**Comment by errordeveloper** [comment]
@gemagomez @martina-if I think this needs to be prioritised for the next release.

File: pkg/addons/default/generate.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
 package defaultaddons
 
+//go:generate curl --silent --location https://github.com/aws/amazon-vpc-cni-k8s/blob/dd631108e61a977809f9b1a1c40232637e734184/config/v1.5/aws-k8s-cni.yaml?raw=1 --output assets/aws-node.yaml
```
Comment (errordeveloper) [inline pkg/addons/default/generate.go]: I can see this was remove in 947cff2907a54ee7e33d15f7cf6432bd76f8a4f2 and it shouldn't had been. Thanks for fixing!

**Comment by errordeveloper** [comment]
🚀

**Comment by jicowan** [comment]
@martina-if @errordeveloper does the aws-node daemonset get overwritten [to 1.5.0] when you deploy a new node group or when you upgrade the cluster control plane with eksctl?

**Pull request merged**